plugins/test/dynamic_test.cc

```diff
@@ -177,14 +199,47 @@ void DynamicTest::TestBody() {
     ASSERT_VM_HEALTH("request_headers", handle, stream);
     CheckPhaseResults("request_headers", invoke.result(), stream, res);
   }
-  if (cfg_.request_body_size() > 0) {
-    for (auto& invoke : *cfg_.mutable_request_body()) {
-      auto res = stream.SendRequestBody(std::move(
-          *absl::WrapUnique(invoke.mutable_input()->release_content())));
-      ASSERT_VM_HEALTH("request_body", handle, stream);
-      CheckPhaseResults("request_body", invoke.result(), stream, res);
+  auto run_body_test = [&handle, &stream, this](
+                           std::string phase,
+                           google::protobuf::RepeatedPtrField<pb::Invocation>
+                               invocations,
+                           auto invoke_wasm) {
+    if (invocations.size() == 0) return;
+    auto body_chunking_plan = cfg_.body_chunking_plan_case();
+    if (invocations.size() != 1 &&
+        body_chunking_plan !=
+            pb::Test::BodyChunkingPlanCase::BODY_CHUNKING_PLAN_NOT_SET) {
+      FAIL()
+          << "Cannot specify body_chunking_plan with multiple body invocations";
+      return;
     }
-  }
+    for (const pb::Invocation& invocation : invocations) {
+      TestHttpContext::Result body_result = TestHttpContext::Result{};
+      auto complete_input_body = ParseBodyInput(invocation.input());
```

Comment:
suggest expanding this `auto` to the actual type, which isn't obvious from the statement by itself.

(See https://engdoc.corp.google.com/eng/doc/devguide/cpp/styleguide.md?cl=head#Type_deduction and go/totw/232 for general recommendations around when to use `auto`)